stdlib/public/stubs/Stubs.cpp

```diff
@@ -327,6 +340,54 @@ extern "C" long double _swift_fmodl(long double lhs, long double rhs) {
     return fmodl(lhs, rhs);
 }
 
+#if defined(__APPLE__)
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_INTERFACE
+extern "C" char ** _swift_get_args(int *outArgLen) {
+  assert(outArgLen != nullptr);
+
+  *outArgLen = *_NSGetArgc();
+  return *_NSGetArgv();
+}
+#elif defined(__linux__) || defined(__CYGWIN__) || defined(__FreeBSD__)
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_INTERFACE
+extern "C" char ** _swift_get_args(int *outArgLen) {
+  assert(outArgLen != nullptr);
+  
+  FILE *cmdline = fopen("/proc/self/cmdline", "rb");
+  if (cmdline == nullptr) {
+    *outArgLen = 0;
+    return nullptr;
```

Comment:
Terminate the process (`swift::fatalError`) or return an empty array.  Returning a null pointer is the worst option, the Swift code will crash.